src/openrct2/ride/Vehicle.cpp

```diff
@@ -5530,64 +5542,64 @@ void Vehicle::UpdateSound()
     switch (vehicleEntry->sound_range)
     {
         case SOUND_RANGE_WHISTLE:
-            screamId = scream_sound_id;
+            currentSound.id = scream_sound_id;
             if (!(gCurrentTicks & 0x7F))
             {
                 if (velocity < 0x40000 || scream_sound_id != OpenRCT2::Audio::SoundId::Null)
-                    goto loc_6D7A97;
+                {
+                    GetLiftHillSound(curRide, currentSound);
+                    break;
+                }
 
                 if ((scenario_rand() & 0xFFFF) <= 0x5555)
                 {
                     scream_sound_id = OpenRCT2::Audio::SoundId::TrainWhistle;
-                    screamVolume = 255;
+                    currentSound.volume = 255;
                     break;
                 }
             }
-            if (screamId == OpenRCT2::Audio::SoundId::NoScream)
-                screamId = OpenRCT2::Audio::SoundId::Null;
-            screamVolume = 255;
+            if (currentSound.id == OpenRCT2::Audio::SoundId::NoScream)
+                currentSound.id = OpenRCT2::Audio::SoundId::Null;
+            currentSound.volume = 255;
             break;
 
         case SOUND_RANGE_BELL:
-            screamId = scream_sound_id;
+            currentSound.id = scream_sound_id;
             if (!(gCurrentTicks & 0x7F))
             {
                 if (velocity < 0x40000 || scream_sound_id != OpenRCT2::Audio::SoundId::Null)
-                    goto loc_6D7A97;
+                {
+                    GetLiftHillSound(curRide, currentSound);
+                    break;
+                }
 
                 if ((scenario_rand() & 0xFFFF) <= 0x5555)
                 {
                     scream_sound_id = OpenRCT2::Audio::SoundId::Tram;
-                    screamVolume = 255;
+                    currentSound.volume = 255;
                     break;
                 }
             }
-            if (screamId == OpenRCT2::Audio::SoundId::NoScream)
-                screamId = OpenRCT2::Audio::SoundId::Null;
-            screamVolume = 255;
+            if (currentSound.id == OpenRCT2::Audio::SoundId::NoScream)
+                currentSound.id = OpenRCT2::Audio::SoundId::Null;
+            currentSound.volume = 255;
             break;
 
         default:
             if ((vehicleEntry->flags & VEHICLE_ENTRY_FLAG_RIDERS_SCREAM))
             {
-                screamId = UpdateScreamSound();
-                if (screamId == OpenRCT2::Audio::SoundId::NoScream)
-                    screamId = OpenRCT2::Audio::SoundId::Null;
-                else if (screamId == OpenRCT2::Audio::SoundId::Null)
-                    goto loc_6D7A97;
+                currentSound.id = UpdateScreamSound();
+                if (currentSound.id == OpenRCT2::Audio::SoundId::NoScream)
+                    currentSound.id = OpenRCT2::Audio::SoundId::Null;
+                else if (currentSound.id == OpenRCT2::Audio::SoundId::Null)
```

Comment:
You can slightly clean this up by not repeating the get lift hill sound. Just rework the ifs so that it doesn't break.

Reply:
Done